app/controllers/posts_controller.rb

```diff
@@ -4,7 +4,7 @@ def index
     @user = User.find(params[:user_id].to_i)
     @no_of_pages = (@user.posts_counter.to_f / 3).ceil
     @current_page = params[:page].nil? || params[:page].to_i < 1 ? 1 : params[:page].to_i
-    @posts = Post.where(author: @user).order(id: :asc).limit(3).offset((@current_page - 1) * 3)
+    @posts = @user.posts.order(id: :asc).limit(3).offset((@current_page - 1) * 3)
```

Comment:
- [ ] Your code needs to address `N + 1` problem when fetching all posts. You can solve solve this problem by modifying this line your _index_ action like this example:
```ruby
@posts = @user.posts.includes(:comments)
```
instead of 
```ruby
@posts = @user.posts

Reply:
@rloterh, I had done this previously but the bullet gem hinted that it was not neccessary and prompted me to remove this. Nonethless, I will go ahead to add it back becuase you've asked me to. Please approve my project.